pkg/clustercache/store.go

```diff
@@ -86,9 +87,16 @@ func (s *GenericStore[Input, Output]) Delete(obj any) error {
 func (s *GenericStore[Input, Output]) GetAll() []Output {
 	s.mutex.RLock()
 	defer s.mutex.RUnlock()
+
+    // Deep copy the stored items to ensure that callers do not modify
+    // the original objects in the store.
 	allItems := make([]Output, 0, len(s.items))
 	for _, item := range s.items {
-		allItems = append(allItems, item)
+		deepCopyable, ok := any(item).(rt.Object);
+		if !ok {
+			continue
+		}
+		allItems = append(allItems, deepCopyable.DeepCopyObject().(Output))
```

Comment:
Minor style change? Matches the code in watchcontroller.go which you've linked to

```suggestion
		if deepCopyable, ok := any(item).(rt.Object); ok {
            allItems = append(allItems, deepCopyable.DeepCopyObject().(Output))
        }
```

Reply:
Don't directly commit this ^^. I think it was formatted using spaces instead of tabs.